packages/vc-handler-ld-local/src/suites/BbsBlsSignature2020.ts

```diff
@@ -0,0 +1,120 @@
+import { blsSign } from '@mattrglobal/node-bbs-signatures'
+import { Bls12381G2KeyPair, BbsBlsSignature2020 as MattrBbsBlsSignature2020 } from '@mattrglobal/jsonld-signatures-bbs'
+import { IKey, TKeyType, VerifiableCredential } from '@veramo/core'
+import { asArray } from '@veramo/utils'
+import suiteContext2020 from 'ed25519-signature-2020-context'
+import * as u8a from 'uint8arrays'
+
+import { SphereonLdSignature } from '../ld-suites'
+
+export class SphereonBbsBlsSignature2020 extends SphereonLdSignature {
+  constructor() {
+    super()
+    // Ensure it is loaded
+    suiteContext2020?.constants
+  }
+
+  getSupportedVerificationType(): string {
+    return 'Bls12381G2Key2020'
+  }
+
+  getSupportedVeramoKeyType(): TKeyType | 'Bls12381G2' {
+    return 'Bls12381G2'
+  }
+
+  getContext(): string {
+    return 'https://w3id.org/security/bbs/v1'
+  }
+
+  getSuiteForSigning(key: IKey, issuerDid: string, verificationMethodId: string): any {
+    const controller = issuerDid
+
+    // DID Key ID
+    const id = verificationMethodId
+    const signer = {
+      sign: async (args: { data: Uint8Array }): Promise<Uint8Array> => {
+        if (!key.privateKeyHex) {
+          throw new Error('privateKey not found!')
+        }
+        const keyPair = {
+          publicKey: new Uint8Array(Buffer.from(key.publicKeyHex)),
+          secretKey: new Uint8Array(Buffer.from(key.privateKeyHex as string)),
+        }
+        const signature = await blsSign({
+          keyPair,
+          messages: [args.data],
+        })
+        return u8a.fromString(`${signature}`)
+      },
+    }
+
+    const options = {
+      id: id,
+      controller: controller,
+      privateKeyBase58: key.privateKeyHex as string,
+      publicKeyBase58: key.publicKeyHex,
+      signer: () => signer,
+      type: this.getSupportedVerificationType(),
+    }
+    /*const key1 = new Bls12381G2KeyPair({
+      publicKeyBase58: hexToMultibase("", MultibaseFormat.BASE58).value
+    });
+    const bbsSignature = new BbsBlsSignature2020({
+      key1
+    })*/
+    let key2: Bls12381G2KeyPair = undefined as unknown as Bls12381G2KeyPair
```

Comment:
i do not think the code here is complete

Reply:
Fixed